testing/mocks/generic.go

```diff
@@ -44,8 +58,9 @@ var (
 	}
 
 	GenericHeader = &flow.Header{
-		ChainID: dps.FlowTestnet,
-		Height:  GenericHeight,
+		ChainID:   dps.FlowTestnet,
+		Height:    GenericHeight,
+		Timestamp: time.Date(1972, 0, 0, 0, 0, 0, 0, time.Local),
```

Comment:
We definitely want to use something without zeros here, so that all parts of the resolution are used. Also, using `time.Local` makes it not deterministic, should probably be `time.UTC`.

Reply:
Unfortunately using `time.UTC` breaks the tests, because when the time is marshaled/unmarshaled the `time.Location` info gets lost 🤔 I looked into it real quick and the only workaround is apparently to use a custom marshaller/unmarshaller so I thought it's not worth the trouble. Also the same goes for the nanoseconds:

When using `time.Date(1972, 11, 12, 13, 14, 15, 16, time.Local)`, the 16 nanoseconds are triggering a diff:

```diff
Diff:
--- Expected
+++ Actual
@@ -12,3 +12,3 @@
  Timestamp: (time.Time) {
-  wall: (uint64) 16,
+  wall: (uint64) 0,
ext: (int64) 62226015255,
```

When using `time.UTC` the `time.Location` does instead:

```diff
Diff:
--- Expected
+++ Actual
@@ -14,3 +14,11 @@
   ext: (int64) 62226018855,
-  loc: (*time.Location)(<nil>)
+  loc: (*time.Location)({
+   name: (string) "",
+   zone: ([]time.zone) <nil>,
+   tx: ([]time.zoneTrans) <nil>,
+   extend: (string) "",
+   cacheStart: (int64) 0,
+   cacheEnd: (int64) 0,
+   cacheZone: (*time.zone)(<nil>)
+  })
},
```

Here is the assertion that causes those errors: `assert.Equal(t, mocks.GenericHeader, got)` in `api/dps/index_test.go`.

Let me know if you know a way around those issues 👍 In the meantime I will only set the other fields and leave 0 nanoseconds and a `time.Local`.

Reply:
As discussed on Slack, you need to use the correct encoding options with ZBOR to both keep full time resolution _and_ time zone information. It's essentially this bit here in our codec:

```go
options.Time = cbor.TimeRFC3339Nano
```